From ccd25f73c955df78d2fb3fe7df445acb497e966e Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 21 Jan 2021 22:25:40 +0200
Subject: [PATCH] [SW-29389] gaudi: security: set psoc etr sec level

PCOS ETR initiator should be using user security level.
Configure it as such.

Change-Id: Ia2bf59992264bc14b1cde277efc4be66c9de53e4
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 4928341..c75b3e2 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -36,6 +36,7 @@
 #include <cpu_if_masks.h>
 #include <psoc_global_conf_masks.h>
 #include <pcie_wrap_masks.h>
+#include <psoc_etr_masks.h>
 #include <mme0_acc_masks.h>
 #include <mme0_sbab_masks.h>
 #include <mme0_ctrl_masks.h>
@@ -748,6 +749,10 @@
 
 	/* Protect GLOBAL_CONF - 4K */
 	preboot_secure_psoc_global_conf();
+
+	/* Set ETR level to user */
+	mask = BIT(PSOC_ETR_AXICTL_PROTCTRLBIT1_SHIFT);
+	regmap_writel(mask, mmPSOC_ETR_AXICTL);
 }
 
 static void preboot_secure_pci_wrap(void)
